cdap-flume/src/main/java/co/cask/cdap/flume/StreamSink.java

```
@@ -0,0 +1,191 @@
+    port = context.getInteger("port", DEFAULT_PORT);
+    sslEnabled = context.getBoolean("sslEnabled", DEFAULT_SSL);
+    authToken = context.getString("authToken");
+    version = context.getString("version", DEFAULT_VERSION);
+    writerPoolSize = context.getInteger("writerPoolSize", DEFAULT_WRITER_POOL_SIZE);
+    streamName = context.getString("streamName");
+    Preconditions.checkState(host != null, "No hostname specified");
+    Preconditions.checkState(streamName != null, "No stream name specified");
+  }
+
+  @Override
+  public Status process() throws EventDeliveryException {
+
+    Status status = Status.READY;
+    Channel channel = getChannel();
+    Transaction transaction = channel.getTransaction();
+
+    try {
+      tryReopenClientConnection();
+      transaction.begin();
+      Event event = channel.take();
+      if (event != null) {
+        writer.write(ByteBuffer.wrap(event.getBody()), event.getHeaders());
+      }
+      transaction.commit();
+
+    } catch (Throwable t) {
+      transaction.rollback();
+      if (t instanceof Error) {
+        throw (Error) t;
+      } else if (t instanceof ChannelException) {
+        LOG.error("Stream Sink {}: Unable to get event from channel {} ", getName(), channel.getName());
+        status = Status.BACKOFF;
+      } else {
+        closeClientQuietly();
+        closeWriterQuietly();
+        throw new EventDeliveryException("Failed to send events", t);
+      }
+    } finally {
+      transaction.close();
+    }
+    return status;
+  }
+
+
+  private void tryReopenClientConnection() throws IOException {
+    if (writer == null) {
+      LOG.debug("Trying to reopen stream writer {} ", streamName);
+      try {
+        createStreamClient();
+      } catch (IOException e) {
+        LOG.error("Error during reopening client by name: {} for host: {}, port: {}. Reason: {} ",
+                  new Object[]{streamName, host, port, e.getMessage(), e});
+        throw new IOException(e);
```

I think you should rethrow the original exception, not a new exception without message